addons/payment_stripe/static/src/js/express_checkout_form.js

```diff
@@ -190,7 +203,11 @@ paymentExpressCheckoutForm.include({
                             detail: carrier.description ? carrier.description:'',
                             amount: carrier.minorAmount,
                         })),
-                        ...this._getOrderDetails(availableCarriers[0].minorAmount),
+                        ...this._getOrderDetails(
+                            availableCarriers[0].minorAmount,
+                            undefined,
+                            adjustedAmountWithFiscalPosition || 0,
+                        ),
```

Comment:
This won't fix the issue; the next call just below overrides it. Also, you're only affecting the display. I would suggest just changing the minor amount in `this.paymentContext['minorAmount']`.